sale_advance_payment/models/sale.py

```diff
@@ -81,10 +83,16 @@ def _compute_advance_payment(self):
                     advance_amount += line_amount
             # Consider payments in related invoices.
             invoice_paid_amount = 0.0
-            for inv in order.invoice_ids:
-                invoice_paid_amount += (
-                    inv.amount_total_signed - inv.amount_residual_signed
-                )
+            for inv in order.invoice_ids.filtered(lambda x: x.state != "cancel"):
+                paid_amount = inv.amount_total_in_currency_signed - inv.amount_residual
+                if inv.currency_id != order.currency_id:
+                    paid_amount = inv.currency_id._convert(
+                        paid_amount,
+                        order.currency_id,
+                        order.company_id,
+                        inv.invoice_date or inv.date,
+                    )
```

Comment:
To avoid computing things if not necessary
```suggestion
                    )
                    else:
                        paid_amount = inv.amount_total_in_currency_signed - inv.amount_residual
```

Reply:
Because potentially, you can do this :

`paid_amount = inv.amount_total_in_currency_signed - inv.amount_residual` even if `inv.currency_id != order.currency_id`